### PR TITLE
Adjust tile bounds so isochrones remain centered

### PR DESCRIPTION
**Pull request by dnesbitt61**
As the max time changes, the isochrone boundaries change and the center location can shift within the grid. This PR tries to adjust the tile boundaries so that the center (first location in the list) remains fixed within a tile/grid cell. This helps minmize "drift" which can cause changes to the contouring.

### CODE REVIEW

File: src/thor/isochrone.cc [hunk 1]
```diff
@@ -68,42 +68,62 @@ void Isochrone::Clear() {
...
-    max_distance = max_seconds * 70.0f * 0.44704f;
+    max_distance = max_seconds * 70.0f * kMPHtoMetersPerSec;
   }
-  shape_interval_ = grid_size * 0.25f;
 
-  // Form grid for isotiles. Convert grid size to degrees.
-  grid_size /= kMetersPerDegreeLat;
-  float lat = origin_locations[0].latlng_.lat();
+  // Set a fixed grid size (in degrees) - about 220m in latitude
+  float grid_size = 0.002f;
```
Comment (kevinkreiser) [inline src/thor/isochrone.cc]: so previously granularity/precision was fixed and now it will be variable based on the latitude?

**Comment by dnesbitt61** [comment]
Not sure I understand. The grids have always been in degrees.

File: src/thor/isochrone.cc (hunk 1 above)
Comment (kevinkreiser) [inline src/thor/isochrone.cc]: oh, i thought the previous code had gridsize in meters and then converted it to degrees based on the lat of the input location but i see that its just divided by a constant. sorry for the misunderstanding

File: valhalla/midgard/gridded_data.h [hunk 2]
```diff
@@ -9,6 +9,10 @@
 namespace valhalla {
 namespace midgard {
 
+// A special generalization value indicating that the application should
+// compute an optimal generalization factor when creating contours.
+constexpr float kOptimalGeneralization = 9999999.0f;
```
Comment (kevinkreiser) [inline valhalla/midgard/gridded_data.h]: might i suggest `std::numeric_limits<float>::max()`

**Comment by kevinkreiser** [comment]
other than comment this looks good to me

**Pull request merged**